x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedConfig.java

```diff
@@ -212,11 +203,9 @@ public DatafeedConfig(StreamInput in) throws IOException {
             this.headers = Collections.emptyMap();
         }
         if (in.getVersion().onOrAfter(Version.CURRENT)) {
-            delayedDataCheckWindow = in.readOptionalTimeValue();
-            shouldRunDelayedDataCheck = in.readBoolean();
+            delayedDataCheckConfig = in.readOptionalWriteable(DelayedDataCheckConfig::new);
         } else {
-            delayedDataCheckWindow = null;
-            shouldRunDelayedDataCheck = false;
+            delayedDataCheckConfig = DelayedDataCheckConfig.disabledDelayedDataCheckConfig();
```

Comment:
I honestly don't know. The default makes sense so that we are consistent with initial creation. I do not fully grok the implications in a mixed versioned cluster however.

Reply:
That makes 2 of us :-) I'd say let's go for the default for the reason you mention.